src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/servicebus/MessageAutoCompletor.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus;
+
+import com.microsoft.azure.servicebus.IMessageReceiver;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class MessageAutoCompletor {
+
+    public static class DeadLetterReason {
+
+        String reason;
+
+        String description;
+
+        public DeadLetterReason(String reason, String description) {
+            this.reason = reason;
+            this.description = description;
+        }
+
+        public DeadLetterReason(String reason) {
+            this(reason, null);
+        }
+    }
+
+    private final BiFunction<UUID, Map<String, Object>, CompletableFuture<Void>> abandonAsyncDelegate;
+
+    private final Function<UUID, CompletableFuture<Void>> completeAsyncDelegate;
+
+    private final BiFunction<UUID, DeadLetterReason, CompletableFuture<Void>> deadLetterAsyncDelegate;
+
+    public MessageAutoCompletor(IMessageReceiver receiver) {
```

Comment:
I'm not sure we'll be able to get the right receiver. We could use the QueueClient instead.

Reply:
Oh I should've checked what QueueClient implements. So it's a `MessageAndSessionPump` interface. But perhaps for clarity can use `IQueueClient` one.